lib/modules/platform/github/index.ts

```diff
@@ -269,8 +269,24 @@ export async function initRepo({
   try {
     let infoQuery = repoInfoQuery;
 
-    if (platformConfig.isGhe) {
+    // GitHub Enterprise Server <3.3.0 doesn't support autoMergeAllowed objects
+    if (
+      platformConfig.isGhe &&
+      // semver not null safe, accepts null and undefined
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      semver.satisfies(platformConfig.gheVersion!, '<3.3.0')
+    ) {
       infoQuery = infoQuery.replace(/\n\s*autoMergeAllowed\s*\n/, '\n');
+    }
+
+    // GitHub Enterprise Server <3.3.0 doesn't support hasIssuesEnabled objects
+    // Logic can be removed, when GHES 3.3 is no longer supported
+    if (
+      platformConfig.isGhe &&
+      // semver not null safe, accepts null and undefined
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+      semver.satisfies(platformConfig.gheVersion!, '<3.3.0')
+    ) {
```

Comment:
Could this be combined with the previous `if`? Seems the same

Reply:
Yeah, I first tried different versions and now opted to pin both to 3.3.x